### PR TITLE
Adding more tests for igDataSource

### PR DESCRIPTION
**Pull request by MayaKirova**
Adding more tests for igDataSource
#63

### CODE REVIEW

File: tests/unit/dataSource/properties/tests.html [hunk 1]
```diff
@@ -184,6 +210,86 @@
...
+		});
+
+		test("Test filtering for customConditions", function () {
+			ds2.clearLocalFilter();
+
+			function filerFunc(val) {
+				return val > 1 ? true : false;
+			};
+		
+			ds2.settings.filtering.customConditions = {
```
Comment (mpavlinov) [inline tests/unit/dataSource/properties/tests.html]: Can you add a test with requireExpr: true, please

File: tests/unit/dataSource/properties/tests.html [hunk 2]
```diff
@@ -552,45 +683,35 @@
...
-				callBackCalled = true;
+		test("Test removeRecordByIndex API method", function () { 
+			var data = [
+				{ "ID": 0, "Name": "Bread", "Price": 2.5 },
+				{ "ID": 1, "Name": "Milk", "Price": 3.5 },
+				{ "ID": 2, "Name": "Vint soda", "Price": 20.9 }
+			];
+			dsLocal.removeRecordByIndex(1, data);
+			equal(data.length, 2, "One item should be removed from the data.");
+			equal(data[data.length - 1].ID, 2, "The last item should be with ID = 2.");
```
Comment (mpavlinov) [inline tests/unit/dataSource/properties/tests.html]: Should you check the transactions collections as well?

File: tests/unit/dataSource/properties/tests.html [hunk 3]
```diff
@@ -493,6 +493,44 @@
...
+				paging: { type: "local", enabled: true, pageSize:1, pageIndex:2 },
+				filtering: { type: "local", defaultFields: [], enabled: true, applyToAllData:true},
+				sorting: { type: "local", defaultFields: [], applyToAllData: true, enabled: true, expressions: [{ fieldName: "ID", dir: "desc" }] }
+			});
+			dsLocalOperations.dataBind();
+			dsLocalOperations.clearLocalFilter();
+
+			//check if paging and sorting are reset
+			equal(dsLocalOperations.pageIndex(),0, "paging is reset");
+			ok(dsLocalOperations.dataView()[0].ID === 2, "Sorting is applied.");
```
Comment (mpavlinov) [inline tests/unit/dataSource/properties/tests.html]: From this check it doesn't seem that sorting is reset.

Comment (MayaKirova) [inline tests/unit/dataSource/properties/tests.html]: Sorting should remain applied. I'll change the comment.

File: tests/unit/dataSource/properties/tests.html (hunk 2 above)
Comment (MayaKirova) [inline tests/unit/dataSource/properties/tests.html]: The transactions are not affected with this method.

**Pull request merged**